word_racer/session.py

```diff
@@ -0,0 +1,295 @@
+        self.round_scores = Counter()
+        self.claims = {}
+        self.valid_words = Counter()
+        self.board = [["_" for _ in range(6)] for __ in range(6)]
+        self.bonus = {}
+        self.nrows = 6
+        self.ncols = 6
+        self.round_finish = False
+        self.reaction_queue = []
+
+    @classmethod
+    def start(cls, ctx):
+        session = cls(ctx)
+        loop = ctx.bot.loop
+        session._task = loop.create_task(session.run())
+        return session
+
+    async def run(self):
+        
+        await self._send_startup_msg()
+
+        # Round loop
+        while self.level < _LEVEL_COUNT:
+            # Round setup
+            self.claims = {}
+            self.valid_words = Counter()
+            self.round_scores = Counter()
+            self._gen_board()
+            self._get_score_dict()
+            self._gen_image()
+            await asyncio.sleep(3)
+
+            # send board image
+            f = discord.File(str(self.output_image_path))
+            await self.ctx.send(f"Starting round {self.level+1}. {len(self.valid_words)} words to find.",file=f)
+
+            # Message handler for round
+            self.round_finish = False
+            await self.run_round()
+
+            # Round cleanup
+            await self.finish_round()
+            self.level += 1
+            
+        await self.end_game()
+
+    async def _send_startup_msg(self):
+        await self.ctx.send("Starting Word Racer. Find words boggle-style and gain points."
+                            " Incorrect calls are -1 point. In rounds 2-4 there are bonuses:"
+                            " blue is 2x points and red is 3x points (they multiplicatively stack)."
+                            " Good luck!")
+
+    async def run_round(self):
+        timer_task = self.ctx.bot.loop.create_task(self.timer_task())
+        reaction_task = self.ctx.bot.loop.create_task(self.reactions_handler())
+        try:
+            await self.ctx.bot.wait_for("message", check=self.check_message, timeout=120)
+        except asyncio.TimeoutError:
+            #Round over
+            pass
+        finally:
+            timer_task.cancel()
+            reaction_task.cancel()
+
+
+    async def reactions_handler(self):
+        while not self.round_finish or self.reaction_queue:
+            if self.reaction_queue:
+                m = self.reaction_queue.pop(0)
+                await m[0].add_reaction(m[1])
+            else:
+                await asyncio.sleep(.25)
+
+    async def finish_round(self):
+        await self.send_round_table(f"- Round {self.level+1} over! \n Round scores:")
+        if self.level in [1,2]:
+            await self.send_table("Total scores so far:")
+        
+        table = ""
+        for word, score in self.valid_words.most_common():
+            claim = "none"
+            if word in self.claims:
+                claim = self.claims[word]
+            table += f"{word}\t{score}\t{claim}\n"
```

Comment:
I suggest handling the initial `{word}\t` spacing with f-string directives instead, as the length of words may vary enough to make the behavior of the tab character unstable.